mautic/mautic
# Allow unicode characters in email addresses

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [X]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [X]
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #...

#### Description:
Creating a new contact with a valid email address that contains a unicode character(for example: jöhn@example.com) results in a sanitized value (Creating the same contact with the API results in an error response).  There is also a possibility to leak personal information by sending an email to the sanitized email. 

The PR contains a change that removes the email sanitization. Therefore we added a length check in the email validator. To allow unicode characters i've added the flag "FILTER_FLAG_EMAIL_UNICODE"  to Mautic\EmailBundle\Helper/EmailValidator::isValidFormat() 

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new contact with an email containing a unicode character (Example: jöhn@example.com)
3. Save the contact and check if the saved email address is the same. 
4. Create a new contact with an email longer than  254 characters
5. Check if there is an error message
